### PR TITLE
Add SonarQube integration to the project

### PR DESCRIPTION
**Pull request by stranljip**
**Motivation:**

* automatically execute checks with SonarQube after each commit in CI

**Modification:**

* modified package.json to execute checks and added the necessary configuration

**Result:**

* SonarQubeChecks are executed automatically

**Additional Remarks:**

This is a new PR after cleaning up my fork and adressing the changes requested in PR #2 (https://github.com/helmholtz-marketplace/helmholtz-marketplace-webapp/pull/2). In detail, I changed the following:

* moved SonarQube access token to a environment variable which is created from a Github secret (see `README.md` for this)
* limited changes in `package.json` to the single added line `"sonarqube-scanner": "^2.6.0"`

I did not change the script in `package.json` starting with `"posttest"` this is valid according to the docs and I tried using `"test:post` which is not executed. 

The comment on the first line in `ci-analysis.js` is necessary, because otherwise eslint complains about extraneous dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,8 @@ HIFIS Marketplace
 ===================
 ![](https://github.com/hifis-cloud/hifis-marketplace/workflows/CI/badge.svg)
 
-An entry point web application for scientific related services from various vendors.
\ No newline at end of file
+An entry point web application for scientific related services from various vendors.
```
Comment (femiadeyemi) [inline README.md:5]: This change in this line is unrelated to this patch.

File: ci-analysis.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const sonarqubeScanner = require('sonarqube-scanner'); // eslint-disable-line import/no-extraneous-dependencies
```
Comment (femiadeyemi) [inline ci-analysis.js:1]: [non-blocking]
why are you keeping this comment, that is
```js
// eslint-disable-line import/no-extraneous-dependencies
```

File: package.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     "test": "karma start",
     "test:coverage": "karma start --coverage",
     "test:watch": "karma start --auto-watch=true --single-run=false",
+    "posttest": "node ci-analysis.js",
```
Comment (femiadeyemi) [inline package.json:12]: I will suggest something like this:
```js
"test:post": "node ci-analysis.js",
```

File: ci-analysis.js (hunk 2 above)
Comment (femiadeyemi) [reply ci-analysis.js:1]: I just notice you comment in the commit message

File: README.md (hunk 1 above)
Comment (femiadeyemi) [reply README.md:5]: ditto

File: package.json (hunk 3 above)
Comment (femiadeyemi) [reply package.json:12]: ditto

File: README.md (hunk 1 above)
Comment (stranljip) [reply README.md:5]: Looking in the code it seems that the change in this line is only the newly added CR/LF, or do I misunderstand your comment?

Comment (femiadeyemi) [reply README.md:5]: I see.

**Comment by femiadeyemi** [comment]
Do you know why the simple github ci test is failing?

**Comment by stranljip** [comment]
> Do you know why the simple github ci test is failing?

Yep. The secret is probably still missing - I do not have the necesssary permissions on this project to add it.

**Comment by femiadeyemi** [comment]
> > Do you know why the simple github ci test is failing?
> 
> Yep. The secret is probably still missing - I do not have the necesssary permissions on this project to add it. I sent it to you in the chat.

Permission granted.
